src/main/java/com/github/dockerjava/netty/handler/HttpResponseStreamHandler.java

```diff
@@ -90,11 +113,19 @@ public int read() throws IOException {
         private void poll() {
             if (readableBytes() == 0) {
                 try {
-                    current = queue.poll(50, TimeUnit.MILLISECONDS);
+                    releaseCurrent();
+                    current = queue.poll(500, TimeUnit.MILLISECONDS);
```

Comment:
There were problems on getting the first data chunk from the queue. I'm looking forward to replace `queue` with `Exchanger`. This code would gone.